src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs

```diff
@@ -524,14 +509,19 @@ private static unsafe int SysReceiveMessageFrom(SafeSocketHandle socket, SocketF
                 return -1;
             }
 
+            if (socketAddressLen == 0)
+            {
+                // We can fail to get peer address on TCP
+                socketAddressLen =  socketAddress.Length;
+                SocketAddressPal.Clear(socketAddress);
+            }
```

Comment:
Is this a bug fix?

Reply:
I'm not sure but possibly yes. On Windows we seems to get the address but it fails on Linux. Perhaps we can also check if this is TCP and get the value from `RemoteEndPoint`? I can write issue for improvement and investigation. 
I had trouble because setting `Size` zero would throw. Alternatively we can allow it to indicate that the buffer does not have any valid address.